### ISSUE TITLE
loss

### ISSUE THREAD

**Issue by ih4cku** [ih4cku/blog#41]
- multinomial logistic loss
- cross entropy loss
- softmax loss

**Comment by ih4cku** [comment]
- http://stats.stackexchange.com/questions/166958/multinomial-logistic-loss-vs-cross-entropy-vs-square-error

**Comment by ih4cku** [comment]
Loss function is to measure the quality of model output.

The typical loss function for classification problem is:

**Comment by ih4cku** [comment]
**Logistic function**: 
![](https://latex.codecogs.com/gif.latex?f%28x%29%3D%5Cfrac%7BL%7D%7B1&plus;e%5E%7B-k%28x-x_0%29%7D%7D)

**Sigmoid function** is a special case of the Logistic function when ![](https://latex.codecogs.com/gif.latex?%5Cinline%20L%3D1%2C%20k%3D1%2C%20x_0%3D0) :
![](https://latex.codecogs.com/gif.latex?%5Csigma%28x%29%3D%5Cfrac%7B1%7D%7B1-e%5E%7B-x%7D%7D)

The output range of the sigmoid function is [0, 1].

The first derivative of sigmoid function is: 
![](https://latex.codecogs.com/gif.latex?%5Cfrac%7B%5Cpartial%5Csigma%7D%7B%5Cpartial%20x%7D%3D%5Csigma%28x%29%281-%5Csigma%28x%29%29)

**Comment by ih4cku** [comment]
In the linear classifier: ![](https://latex.codecogs.com/gif.latex?f%28x%3BW%2Cb%29%3DWx&plus;b), each **row** of W is a classifier. Each row of W can corresponds to a **template** of the category. So, the classifier is doing **template matching**.
